pkgs/tools/misc/digitemp/default.nix

```diff
@@ -0,0 +1,68 @@
+{ fetchFromGitHub
+, gnumake
+, lib
+, libusb
+, stdenv
+
+, defaultAdapter ? "DS9097U"
+# that support seems to be broken
+, ds2490 ? false
+}:
+
+stdenv.mkDerivation rec {
+  pname = "digitemp";
+  version = "3.7.2";
+
+  src = fetchFromGitHub {
+    owner = "bcl";
+    repo = "digitemp";
+    rev = "v${version}";
+    sha256 = "19zka5fcdxhhginaspak76l984iqq9v2j6qrwvi5mvca7bcj8f72";
+  };
+
+  nativeBuildInputs = [ gnumake ];
+  buildInputs = lib.optional ds2490 libusb;
+
+  buildPhase = ''
+    runHook preBuild
+    make clean
+    make -j$NIX_BUILD_CORES LOCK=no ds9097 ds9097u ${lib.optionalString ds2490 "ds2490"}
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    install -D -t $out/bin digitemp_*
+  '' + (lib.optionalString (defaultAdapter != "") ''
+    pushd $out/bin
+    ln -s "digitemp_${defaultAdapter}" digitemp
+    popd
+  '') + ''
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Temperature logging and reporting using Maxim's iButtons and 1-Wire protocol";
+    longDescription = ''
+      DigiTemp is a command line application used for reading 1-wire sensors like
+      the DS18S20 temperature sensor, or DS2438 battery monitor. DigiTemp supports
+      the following devices:
+
+        DS18S20 (and older DS1820) Temperature Sensor
+        DS18B20 Temperature Sensor
+        DS1822 Temperature Sensor
+        DS2438 Battery monitor
+        DS2409 1-wire coupler (used in 1-wire hubs)
+        DS2422 Counter
+        DS2423 Counter
+
+      The output format can be customized and all settings are stored in a
+      configuration file (.digitemprc) in the current directory. DigiTemp can
+      repeatedly read the sensors and output to stdout and/or to a logfile.
+    '';
+    homepage = "https://www.digitemp.com";
+    license = licenses.gpl2Plus;
+    maintainers = with maintainers; [ zseri ];
+    platforms = platforms.all;
```

Comment:
I wasn't able to figure out a fitting `meta.platforms` list... it supports: at least Linux on x86 (32bit and 64bit, marked as stable in Gentoo), cygwin (last version with support for that), BSD, Darwin, AIX, ..., almost everything which has or can emulate device files... so that excludes Javascript.

Reply:
setting it to all here is fine.